File: projects/Go/examples/create.go
package main

import "fmt"
import "../proto/fbe"
import "../proto/proto"

func main() {
	// Create a new account using FBE model
	var account = proto.NewAccountModel(fbe.NewEmptyBuffer())
	modelBegin := account.CreateBegin()
	accountBegin, _ := account.Model().SetBegin()
	_ = account.Model().Id.Set(1)
	_ = account.Model().Name.Set("Test")
	_ = account.Model().State.SetValue(proto.State_good)
	walletBegin, _ := account.Model().Wallet.SetBegin()
	_ = account.Model().Wallet.Currency.Set("USD")
	_ = account.Model().Wallet.Amount.Set(1000.0)
	account.Model().Wallet.SetEnd(walletBegin)
	account.Model().SetEnd(accountBegin)
	account.CreateEnd(modelBegin)
	if ok := account.Verify(); !ok {
		panic("verify error")
	}

	// Show the serialized FBE size
	fmt.Printf("FBE size: %d\n", account.Buffer().Size())

	// Access the account using the FBE model
	access := proto.NewAccountModel(account.Buffer())
	if ok := access.Verify(); !ok {
		panic("verify error")
	}

	accountBegin, _ = access.Model().GetBegin()
	id, _ := access.Model().Id.Get()
	name, _ := access.Model().Name.Get()
	state, _ := access.Model().State.Get()
	walletBegin, _ = access.Model().Wallet.GetBegin()
	walletCurrency, _ := access.Model().Wallet.Currency.Get()
	walletAmount, _ := access.Model().Wallet.Amount.Get()
	access.Model().Wallet.GetEnd(walletBegin)
	access.Model().GetEnd(accountBegin)

	// Show account content
	fmt.Println()
	fmt.Printf("account.id = %v\n", id)
	fmt.Printf("account.name = %v\n", name)
	fmt.Printf("account.state = %v\n", state)
	fmt.Printf("account.wallet.currency = %v\n", walletCurrency)
	fmt.Printf("account.wallet.amount = %v\n", walletAmount)
}

File: projects/Ruby/examples/create.rb
require_relative '../proto/proto'

# Create a new account using FBE model
account = Proto::AccountModel.new(FBE::WriteBuffer.new)
model_begin = account.create_begin
account_begin = account.model.set_begin
account.model.id.set(1)
account.model.name.set('Test')
account.model.state.set(Proto::State.good)
wallet_begin = account.model.wallet.set_begin
account.model.wallet.currency.set('USD')
account.model.wallet.amount.set(1000.0)
account.model.wallet.set_end(wallet_begin)
account.model.set_end(account_begin)
account.create_end(model_begin)
raise "Verify error!" unless account.verify

# Show the serialized FBE size
puts "FBE size: #{account.buffer.size}"

# Access the account using the FBE model
access = Proto::AccountModel.new(FBE::ReadBuffer.new)
access.attach_buffer(account.buffer)
raise "Verify error!" unless access.verify

account_begin = access.model.get_begin
id = access.model.id.get
name = access.model.name.get
state = access.model.state.get
wallet_begin = access.model.wallet.get_begin
wallet_currency = access.model.wallet.currency.get
wallet_amount = access.model.wallet.amount.get
access.model.wallet.get_end(wallet_begin)
access.model.get_end(account_begin)

# Show account content
puts
puts "account.id = #{id}"
puts "account.name = #{name}"
puts "account.state = #{state}"
puts "account.wallet.currency = #{wallet_currency}"
puts "account.wallet.amount = #{wallet_amount}"
